pkg/controller/job/jobcontroller.go

```diff
@@ -167,14 +171,32 @@ func (jm *JobController) addPod(obj interface{}) {
 		jm.deletePod(pod)
 		return
 	}
-	if job := jm.getPodJob(pod); job != nil {
+
+	// If it has a ControllerRef, that's all that matters.
+	if controllerRef := controller.GetControllerOf(pod); controllerRef != nil {
+		if controllerRef.Kind != controllerKind.Kind {
+			// It's controlled by a different type of controller.
+			return
+		}
+		job, err := jm.jobLister.Jobs(pod.Namespace).Get(controllerRef.Name)
+		if err != nil {
+			return
+		}
 		jobKey, err := controller.KeyFunc(job)
 		if err != nil {
-			utilruntime.HandleError(fmt.Errorf("Couldn't get key for job %#v: %v", job, err))
```

Comment:
In a comment on another ControllerRef PR, the reviewer asked me to remove this type of logging during watch handlers.